sat/apiclient/vcs.py

```diff
@@ -133,12 +170,26 @@ def get_commit_hash_for_branch(self, branch):
         target_ref = f'refs/heads/{branch}'
         return self.remote_refs.get(target_ref)
 
-    @cached_property
-    def vcs_host(self):
-        """str: Hostname of the VCS server."""
-        return 'api-gw-service-nmn.local'  # TODO: Update per CRAYSAT-898.
+    def clone(self, branch=None, directory=None, single_branch=False):
+        """Clone the VCS repo to the given path.
 
-    @cached_property
-    def clone_url(self):
-        """str: a full, git-clone-able URL to the repository"""
-        return urlunparse(('https', self.vcs_host, self.repo_path, '', '', ''))
+        Args:
+            branch (str, optional): the branch to set as HEAD in the cloned repo.
+                If omitted, defaults to the HEAD of the remote repo.
+            directory (str, optional): the directory to clone the repo to.
+                If omitted, clone to a directory matching the name of the repo.
+            single_branch (bool, optional): whether to clone just a single branch.
+        """
+        git_cmd = ['git', 'clone']
```

Comment:
Would it make sense to put `--depth=1` in the command options?

Reply:
I think I'll add it as an argument to this function, and I'll use it in `HPCSoftwareRecipe.clone_content` since we really do only need the tip of that branch in that case.